### PR TITLE
🌱 Improve vm operator target + use config/wcp

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Improving the vm-operator targets to 
- standardize target naming
- movr vm-operator.tmp folder away from root
- move kustomize.yaml and patches to a config foldr under test/infrastructure/vm-operator

Also, start using vm-operator/config/wcp instead of config/local as suggested by vm-operator's maintainers

### CODE REVIEW

File: test/infrastructure/vm-operator/config/vm-operator-env-var-patch.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This patch derives from vm-operator config/wcp/vmoperator/manager_env_var_patch.yaml, but with
+# - NETWORK_PROVIDER = true and NETWORK_PROVIDER = VSPHERE_NETWORK as suggested by the vm-operator maintainers.
```
Comment (sbueringer) [inline test/infrastructure/vm-operator/config/vm-operator-env-var-patch.yaml:2]:
```suggestion
# - VSPHERE_NETWORKING = true and NETWORK_PROVIDER = VSPHERE_NETWORK as suggested by the vm-operator maintainers.
```
I think

File: test/infrastructure/vm-operator/config/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+# Set hostNetwork: false because we are running out of the supervisor
+# Note: this requires also to drop KUBERNETES_SERVICE_HOST/PORT variables, which is implemented in a patch below.
+- vm-operator-host-network.yaml
+
+# Drop VC cert bundle because we are running out of the supervisor
+- vm-operator-drop-certs-volume.yaml
+
+# This patch derives from vm-operator config/wcp/vmoperator/manager_env_var_patch.yaml, but with
+# - NETWORK_PROVIDER = true and NETWORK_PROVIDER = VSPHERE_NETWORK as suggested by the vm-operator maintainers.
```
Comment (sbueringer) [inline test/infrastructure/vm-operator/config/kustomization.yaml:37]:
```suggestion
# - VSPHERE_NETWORKING = true and NETWORK_PROVIDER = VSPHERE_NETWORK as suggested by the vm-operator maintainers.
```

File: Makefile [hunk 3]
```diff
@@ -799,10 +799,13 @@ set-manifest-image:
 ##@ vm-operator:
 
 .PHONY: release-vm-operator
-release-vm-operator: docker-vm-operator-build-all vm-operator-manifest-build docker-vm-operator-push-all ## Build and push the vm-operator image and manifest for usage in CI
+release-vm-operator: docker-build-all-vm-operator generate-manifests-vm-operator docker-push-all-vm-operator clean-vm-operator ## Build and push the vm-operator image and manifest for usage in CI
 
-.PHONY: vm-operator-checkout
-vm-operator-checkout:
+.PHONY: release-local-vm-operator
+release-local-vm-operator: docker-build-all-vm-operator generate-manifests-vm-operator clean-vm-operator ## Build and push the vm-operator image and manifest for local usage only
```
Comment (sbueringer) [inline Makefile:805]:
```suggestion
release-local-vm-operator: docker-build-all-vm-operator generate-manifests-vm-operator clean-vm-operator ## Build the vm-operator image and manifest for local usage only
```
The diff seems to be that we don't push.

Q: Do you want to name this target release-vm-operator-local? (no strong opinion from my side)

File: test/infrastructure/vm-operator/vm-operator-v1.8.6-0-gde75746a.yaml [hunk 4]
```diff
@@ -7479,71 +7478,6 @@ spec:
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
-metadata:
```
Comment (sbueringer) [inline test/infrastructure/vm-operator/vm-operator-v1.8.6-0-gde75746a.yaml:7482]: I assume we don't need this one

**Comment by chrischdi** [comment]
lgtm pending stefans comments :-)

**Comment by sbueringer** [comment]
Thx!

/lgtm

**Review by chrischdi — COMMENTED** [review]
/approve

**Comment by sbueringer** [comment]
@fabriziopandini should this be backported for 1.10?

**Pull request merged**